pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/KitRequestShipping.java

```diff
@@ -685,45 +685,45 @@ public static Map<String, List<KitRequestShipping>> getAllKitRequestsByRealm(@No
-                                String apiKey = DSMServer.getDDPEasypostApiKey(ddpInstance.getName());
-                                if (StringUtils.isNotBlank(apiKey) && kit.getEasypostAddressId() != null && StringUtils.isNotBlank(
-                                        kit.getEasypostAddressId())) {
-                                    getAddressPerEasypost(ddpInstance, kit, apiKey);
-                                } else {
-                                    if (participantsESData != null && !participantsESData.isEmpty()) {
-                                        ddpParticipant = ElasticSearchUtil.getParticipantAsDDPParticipant(participantsESData, key);
-                                        if (ddpParticipant != null) {
-                                            kit.setParticipant(ddpParticipant);
+                if (StringUtils.isNotBlank(ddpInstance.getParticipantIndexES())) {
+
+                    Map<String, Map<String, Object>> participantsESData =
+                            ElasticSearchUtil.getDDPParticipantsFromES(ddpInstance.getName(), ddpInstance.getParticipantIndexES());
+
+                    for (String key : kitRequests.keySet()) {
+                        List<KitRequestShipping> kitRequest = kitRequests.get(key);
+                        DDPParticipant ddpParticipant = null;
+                        boolean checkedParticipant = false;
+
+                        for (KitRequestShipping kit : kitRequest) {
+                            if (StringUtils.isNotBlank(kit.getRealm())) {
+                                if (participantsESData != null && !participantsESData.isEmpty()) {
+                                    kit.setPreferredLanguage(ElasticSearchUtil.getPreferredLanguage(participantsESData, key));
+                                }
+                                // ERROR need address; QUEUE need name label if realm = RGP
+                                // UPLOADED and DEACTIVATED and TRIGGERED and WAITING need shortId if getCollaboratorParticipantId is blank
+                                if ((ERROR.equals(target) || ((QUEUE.equals(target) || UPLOADED.equals(target)) && ddpInstance.isHasRole()))
+                                        || (
+                                        (UPLOADED.equals(target) || DEACTIVATED.equals(target) || TRIGGERED.equals(target) || OVERVIEW.equals(
+                                                target) || WAITING.equals(target))
+                                                && StringUtils.isBlank(kit.getBspCollaboratorParticipantId()))) {
+                                    String apiKey = DSMServer.getDDPEasypostApiKey(ddpInstance.getName());
+                                    if (StringUtils.isNotBlank(apiKey) && kit.getEasypostAddressId() != null && StringUtils.isNotBlank(
+                                            kit.getEasypostAddressId())) {
+                                        getAddressPerEasypost(ddpInstance, kit, apiKey);
```

Comment:
I think this is still needed even if we do not get the info from ES.

Reply:
@SimoneMaiwald I refactored it to make it look nicer and keep all of the data that we dont get from participants. Let me know if this works